gazeplay-games/src/test/java/net/gazeplay/games/mediaPlayer/GazeMediaPlayerTest.java

```
@@ -0,0 +1,53 @@
+package net.gazeplay.games.mediaPlayer;
+
+import javafx.scene.control.Button;
+import javafx.scene.image.ImageView;
+import net.gazeplay.components.StackPaneButton;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.runner.RunWith;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.testfx.framework.junit5.ApplicationExtension;
+
+@ExtendWith(ApplicationExtension.class)
+@RunWith(MockitoJUnitRunner.class)
+public class GazeMediaPlayerTest {
+
+    @BeforeEach
+    public void initMocks() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void shouldCreateButtonOfSimpleStackPaneButton() {
+        StackPaneButton stackPaneButton = GazeMediaPlayer.createSimpleStackPaneButton(500, 500);
+        assert stackPaneButton.getButton() != null;
+    }
+
+    @Test
+    void shouldCreateButtonOfTextStackPaneButton() {
+        StackPaneButton stackPaneButton = GazeMediaPlayer.createTextStackPaneButton("Test", 500, 500);
+        assert stackPaneButton.getButton() != null;
+        assert stackPaneButton.getButton().getText().equals("Test");
+    }
+
+    @Test
+    void shouldCreateButtonOfGraphicStackPaneButton() {
+        try {
+            StackPaneButton stackPaneButton = GazeMediaPlayer.createGraphicStackPaneButton(500, 500, "src/test/resources/images/blue/blue.jpg");
+            ImageView graphicImageView = (ImageView) stackPaneButton.getButton().getGraphic();
+            assert stackPaneButton.getButton() != null;
+            assert graphicImageView.getImage().getUrl().contains("blue.jpg");
+        } catch (Exception ignored) {
+
+        }
```

I actually have a problem here @Tohaker , if I'm not adding this try catch I will have an IllegalArgumentException, I don't really get why...

---

You could just make the test throw the IllegalArgumentException, thats fine for Tests. The test runner will print the stacktrace if that test fails, so you can debug it later.

---

Ok, but then I have a :
> Task :gazeplay-games:test FAILED

net.gazeplay.games.mediaPlayer.GazeMediaPlayerTest > shouldCreateButtonOfGraphicStackPaneButton() FAILED
    java.lang.IllegalArgumentException at GazeMediaPlayerTest.java:38
        Caused by: java.lang.IllegalArgumentException at GazeMediaPlayerTest.java:38

24 tests completed, 1 failed

FAILURE: Build failed with an exception.

I don't understand where could be the "IllegalArgument"

---

Ok it was actually a question of path, I'm solving the conv

---

You shouldn't swallow exceptions in tests if the test fails without it. Either the test is not being created properly, or your tested method is not working.